app/src/main/java/org/piwigo/ui/login/LoginViewModel.java

```diff
@@ -219,32 +216,48 @@ public void onCompleted() {
         }
 
         @Override
-        public void onError(Throwable e) {
+        public void onError(Throwable e)  {
             Log.e(TAG, e.getMessage());
+            e.printStackTrace();
             loginError.setValue(e);
+         }
```

Comment:
I don't catch here how you handle the HTTP-fallback, in case https doesn't work

Reply:
I don't. It's a very bad idea IMO - because the error message is the one of http, which might not even exit - even if I configure explicitiy https.

So I expand the url in the UI, so the user sees what is configured (if not given a protocol) and the user can change if it meant http.

Reply:
Somehow this is how I personally also would prefer it, especially because it accepts a user given protocol without blindly changing. On the other side piwigo team faces problems with non-technical users who do not self-host the gallery and they even don't get the difference, so there automatically switching to https (if possible) is a good idea, but we cannot expect that they recognize (or understand) what the automatic https-prefixing in the UI technically means and why it doesn't work :-(

Reply:
ok, I had another discussion with @plegall and you solution is good, if we add one more action:
- if https connection fails a snackbar is shown with the error message. There we should add an action to "try unencrypted" which exchanges https by http in the UI, while the conenction always uses exactly what is in the text field.

with that we are secure by default and do not transmit credentials unencrypted (what is there today with the automatic try of http if https fails for some reason) and even the not-so-advanced users with a piwigo setup in a local network (where http only can be acceptable) will have an esay way to connect.

Reply:
sounds good